.github/workflows/cd.yml

```diff
@@ -0,0 +1,67 @@
+name: Continuous Deployment
+
+on:
+  push:
+    branches: [main]
+  workflow_dispatch:
+
+defaults:
+  run:
+    shell: bash
+
+concurrency:
+  group:  ${{ github.workflow }}-${{ github.head_ref || github.ref }}
```

Comment:
and this?

Reply:
By default, GitHub Actions allows multiple jobs within the same workflow, multiple workflow runs within the same repository, and multiple workflow runs across a repository owner's account to run concurrently. This line stops (cancels) a previous workflow if the user pushes again on the same branch/PR. IMO this is valuable since it reduces the load on the runners. Typically you don't care if the GH actions of your previous commit didn't run though if you for example submitted a bug fix commit later